python/tests/integration/test_worker_recycling_integration.py

```
@@ -0,0 +1,298 @@
+        # resets a worker to ~baseline; without recycling this stays high.
+        min_rss_late = float("inf")
+
+        def sample() -> None:
+            nonlocal peak_rss, min_rss_late
+            seen_workers.update(child_pids(server.process.pid))
+            current = [
+                get_rss_bytes(pid) for pid in child_pids(server.process.pid)
+            ]
+            live = [rss / MIB for rss in current if rss]
+            if live:
+                peak_rss = max(peak_rss, max(live))
```

Ruff (0.15.20)

[warning] Nested `max` calls can be flattened

Flatten nested `max` calls

(PLW3301)